python/templates/macros/iterator.jinja2

```diff
@@ -3,10 +3,20 @@
 {% set ptr_init = 'podio::utils::MaybeSharedPtr<' + class.bare_type +'Obj>{nullptr}' %}
 class {{ iterator_type }} {
 public:
+  using value_type = {{ class.bare_type }};
+  using difference_type = ptrdiff_t;
+  using reference = {{ prefix }}{{ class.bare_type }};
+  using pointer = {{ prefix }}{{ class.bare_type }}*;
+  using iterator_category = std::input_iterator_tag;
+
   {{ iterator_type }}(size_t index, const {{ class.bare_type }}ObjPointerContainer* collection) : m_index(index), m_object({{ ptr_init }}), m_collection(collection) {}
+  {{ iterator_type }}(): {{ iterator_type }}(0, nullptr) {}
 
-  {{ iterator_type }}(const {{ iterator_type }}&) = delete;
-  {{ iterator_type }}& operator=(const {{ iterator_type }}&) = delete;
+  {{ iterator_type }}(const {{ iterator_type }}&) = default;
+  {{ iterator_type }}({{ iterator_type }}&&) = default;
+  {{ iterator_type }}& operator=(const {{ iterator_type }}&) = default;
+  {{ iterator_type }}& operator=({{iterator_type}}&&) = default;
```

Comment:
I suppose all of these are working "as expected" if they are `defaulted`? In the end all the members are either pointers into some other collection or values that behave as expected, so I think it should be OK, just to make sure

Reply:
Normally we would go with rule 0 and not define anything, but there is a pointer member so this time we should be explicit about the semantics we want. Our pointer is not-owning so we want just the simplest assignment which is exactly what `default` does